tests/foreman/cli/test_host.py

```diff
@@ -843,14 +844,20 @@ def test_positive_list_infrastructure_hosts(
     Host.update({'name': default_sat.hostname, 'new-organization-id': module_org.id})
     # list satellite hosts
     hosts = Host.list({'search': 'infrastructure_facet.foreman=true'})
-    assert len(hosts) == 1
+    if is_open('BZ:1994685'):
+        assert len(hosts) == 2
+    else:
+        assert len(hosts) == 1
```

Comment:
I might try to delete the ghost host with `if BZ open` condition here if it's easy to identify that host for deletion. And then will stick to hosts count assertion to 1.

Not a blocking comment though!